joomla/joomla-cms
# [4.0] Feature: front-end users/user view

This PR add a new feature to **display a list of Users** from a **selected User Group**. You can use this new view for example to display a list of Users from the User Group "Author". 

### Summary of Changes
The front-end did not have a model/view to display a list of Users from a selected User Group. The back-end model of com_users has some authentication so that you cannot get data from the Users table if you are not logged in. 

This PR adds a Front-end View + Model for Users and User.
It adds a new menu item type In the back-end: 

### Testing Instructions
Create some Users that you assign to some User Group (e.g. Author and Editor).
Add some Articles that you assign to those new Users.

In Menus > Main Menu, create a New Menu Item of type "Users: List Users from a Group"

![users-backend1](https://user-images.githubusercontent.com/1217850/57194704-11e51680-6f4a-11e9-8f8a-b85670f548e5.png)

Under the "Users" tab, configure what the new menu item should display:
- User Groups: select a few
- Exclude Users: select a few
- Select if you want to see a Header or not
- Configure if you want to see the Number of Articles by that User
- Configure if you want to see the Last Visit date of that User
- Configure the Date Format if you want to show it differently
- Configure if you want to see the Online status of that User

![users-backend2](https://user-images.githubusercontent.com/1217850/57194703-11e51680-6f4a-11e9-89a0-daa46bdf97a1.png)

### Expected result

#### Front-end view: Users
![users-front-end1](https://user-images.githubusercontent.com/1217850/57194702-11e51680-6f4a-11e9-9d8b-589ef7442eaa.png)

#### Front-end view: Single User
![users-front-end2](https://user-images.githubusercontent.com/1217850/57194701-11e51680-6f4a-11e9-92a7-ace728bbb625.png)
